blog/2020-04-09-debezium-apicurio-registry.adoc

```diff
@@ -0,0 +1,636 @@
+Please note that the endpoint does not reference the native API, but the compatibility layer so even if `kafkacat` is not yet aware of the new API, it can easily interoperate with Apicurio.
+
+The resulting message should look like:
+
+[source,json]
+----
+{
+  "before": null,
+  "after": {
+    "Value": {
+      "id": 1001,
+      "first_name": "Sally",
+      "last_name": "Thomas",
+      "email": "sally.thomas@acme.com"
+    }
+  },
+  "source": {
+    "version": "1.1.0.Final",
+    "connector": "mysql",
+    "name": "dbserver1",
+    "ts_ms": 0,
+    "snapshot": {
+      "string": "true"
+    },
+    "db": "inventory",
+    "table": {
+      "string": "customers"
+    },
+    "server_id": 0,
+    "gtid": null,
+    "file": "mysql-bin.000003",
+    "pos": 154,
+    "row": 0,
+    "thread": null,
+    "query": null
+  },
+  "op": "c",
+  "ts_ms": {
+    "long": 1586336163386
+  },
+  "transaction": null
+}
+----
+
+In this case we get only message payload without the schema identifier but we can still query the registry using schema name:
+
+[source,bash]
+----
+$ docker run --rm --tty   --network tutorial_default debezium/tooling bash -c 'http http://apicurio:8080/artifacts/dbserver1.inventory.customers-value | jq .'
+----
+
+The resulting schema description is slightly different for the previous ones as it has an Avro flavour:
+
+[source,json]
+----
+{
+  "type": "record",
+  "name": "Envelope",
+  "namespace": "dbserver1.inventory.customers",
+  "fields": [
+    {
+      "name": "before",
+      "type": [
+        "null",
+        {
+          "type": "record",
+          "name": "Value",
+          "fields": [
+            {
+              "name": "id",
+              "type": "int"
+            },
+            {
+              "name": "first_name",
+              "type": "string"
+            },
+            {
+              "name": "last_name",
+              "type": "string"
+            },
+            {
+              "name": "email",
+              "type": "string"
+            }
+          ],
+          "connect.name": "dbserver1.inventory.customers.Value"
+        }
+      ],
+      "default": null
+    },
+    {
+      "name": "after",
+      "type": [
+        "null",
+        "Value"
+      ],
+      "default": null
+    },
+...
+  ],
+  "connect.name": "dbserver1.inventory.customers.Envelope"
+}
+----
+
+The connector registration request also differs from the standard one in a handful of lines:
+
+[source,json]
+----
+...
+"key.converter": "io.apicurio.registry.utils.converter.AvroConverter", <1>
+"key.converter.apicurio.registry.url": "http://apicurio:8080", <2>
+"key.converter.apicurio.registry.converter.serializer":
+    "io.apicurio.registry.utils.serde.AvroKafkaSerializer", <3>
+"key.converter.apicurio.registry.converter.deserializer":
+    "io.apicurio.registry.utils.serde.AvroKafkaDeserializer", <3>
```

Comment:
Does it actually need the *de*serializer configuration for a source connector? Same for value below.

Reply:
It needs as they are instantiated in `configure` method, not lazily when/if needed.

Reply:
Ok, gotcha. Why is it that the converter needs an explicit serializer config anyways? I.e. which (de)serializer should the `AvroConverter` use, if not the Avro ones?

Reply:
It is pluggable so you can use alternative serde implementations if you wish.
Unfortunately there is no sensible default set

Reply:
Ok; logged an issue with Apicurio: https://github.com/Apicurio/apicurio-registry/issues/378.